R/qExportWig.R

```diff
@@ -355,7 +359,29 @@ qExportWig <- function(proj,
         tmp <- Rsamtools::scanBamHeader(bamfiles[[1]][1])[[1]]$targets
         si <- GenomeInfoDb::Seqinfo(names(tmp), tmp)
     }
-    lapply(seq_len(n), function(i) {
+    
+    # digest clObj
+    clparam <- getListOfBiocParallelParam(clObj)
+    nworkers <- unlist(lapply(clparam, BiocParallel::bpworkers))
+    # will use only one layer of parallelization, select best
+    clsel <- which.max(nworkers) 
+    if (!inherits(clparam[[clsel]], c("BatchJobsParam", "SerialParam"))) { 
+        # don't test loading of QuasR on current or BatchJobs cluster nodes
+        message("preparing to run on ", min(n, nworkers[clsel]),
+                " ", class(clparam[[clsel]]), " nodes...", appendLF = FALSE)
+        ret <- BiocParallel::bplapply(seq.int(nworkers[clsel]), 
+                                      function(i) library(QuasR), 
+                                      BPPARAM = clparam[[clsel]])
+        if (!all(vapply(ret, function(x) "QuasR" %in% x, TRUE)))
+            stop("'QuasR' package could not be loaded on all nodes")
+        message("done")
+    }
+    
+    # avoid nested parallelization
+    if (!inherits(clparam[[clsel]], "SerialParam"))
```

Comment:
Is something here using OpenMP so that it is necessary to limit OMP threads, or is this just a precaution? In any case, it does not hurt.